### PR TITLE
[Driver] Correct comment on default for -falign-functions

### PR DESCRIPTION
**Pull request by pogo59**

### CODE REVIEW

File: clang/lib/Driver/ToolChains/CommonArgs.cpp [hunk 1]
```diff
@@ -1982,7 +1982,7 @@ unsigned tools::ParseFunctionAlignment(const ToolChain &TC,
     return 0;
 
   if (A->getOption().matches(options::OPT_falign_functions))
-    return 0;
+    return 4; // log2(16)
```
Comment (compnerd) [inline clang/lib/Driver/ToolChains/CommonArgs.cpp:1985]: This is not true. Quoting the documentation:

```
If n is not specified or is zero, use a machine-dependent default. The maximum allowed n option value is 65536.
```

16 is a reasonable default for X86, but I don't think that it universally makes sense. Can we switch based on the architecture and continue to return a default of `0`?

Comment (pogo59) [reply clang/lib/Driver/ToolChains/CommonArgs.cpp:1985]: Ah, I should have said "the commentary" on ParseFunctionAlignment, which does say that no argument means 16.
Which documentation are you quoting? I don't see this option described in [the clang user manual](https://clang.llvm.org/docs/UsersManual.html).

I'm not aware that the driver, or clang generally, knows about preferred alignment by target, but I guess we can invent a parameter for that.

Comment (MaskRay) [reply clang/lib/Driver/ToolChains/CommonArgs.cpp:1985]: I agree that clangDriver should not hard code a value like 16.

The relevant code is at `llvm/lib/CodeGen/MachineFunction.cpp`

```cpp
  // FIXME: Shouldn't use pref alignment if explicit alignment is set on F.
  // FIXME: Use Function::hasOptSize().
  if (!F.hasFnAttribute(Attribute::OptimizeForSize))
    Alignment = std::max(Alignment,
                         STI->getTargetLowering()->getPrefFunctionAlignment());
```

**Comment by pogo59** [comment]
The driver can't really be poking down into LLVM's TargetLowering; at best it could be querying Clang's TargetInfo, which knows a lot about data alignment but does not currently have any knowledge of code alignment. So, I've reverted the change and instead fixed the incorrect comment.

**Review by compnerd — APPROVED** [review]

**Review by MaskRay — APPROVED** [review]
LGTM.

The patch title/description need to be changed to ensure "Squash and merge" gets the right message (the git commit message is ignored).

**Pull request merged**